### ISSUE TITLE
Feature request: Make reconnection logic configureable

### ISSUE THREAD

**Issue by coonmoo** [livekit/client-sdk-js#261]
First of all thanks for the great work here!

We have a use-case where the client can be disconnected from the network for more than 60 seconds.
In this case the livekit client gives up reconnecting and stops all tracks.

Instead of hardcoding it here: https://github.com/livekit/client-sdk-js/blob/a492d213ac9c2e20caaf7147cb4930c65bff839b/src/room/RTCEngine.ts#L28

I would like to be able to configure it myself & specify if the tracks should be stopped.
[Example how SignalR handles that.](https://github.com/dotnet/aspnetcore/blob/8b30d862de6c9146f466061d51aa3f1414ee2337/src/SignalR/clients/ts/signalr/src/DefaultReconnectPolicy.ts)

I'm looking for something like this

```
export class MyReconnectPolicy implements IRetryPolicy {
   public nextRetryDelayInMilliseconds(retryContext: RetryContext): number | null {
      if (retryContext.previousRetryCount > 10) {
         retryContext.stopTracks = false;
         return null;
      }

      return retryContext.previousRetryCount * 1000;
   }
}

```

```
this.room = new Room({
 reconnectPolicy: new MyReconnectPolicy()
});
```

**Comment by davidzhao** [comment]
yeah, this is a great suggestion! we could provide a few common reconnect policies in the SDK while making it easy for users to override that behavior.

are you interested in creating a PR for this?
